packages/web3-eth/src/index.ts

```diff
@@ -0,0 +1,371 @@
+					response,
+					// TODO
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-expect-error
+					convertibleTransactionInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	public async getTransactionReceipt<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		transactionHash: HexString32Bytes,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionReceipt(
+			this._requestManager,
+			transactionHash,
+		);
+
+		return response === null
+			? response
+			: convertObjectPropertiesToValidType(
+					response,
+					convertibleReceiptInfoProperties,
+					returnType ?? this._options.defaultReturnType,
+			  );
+	}
+
+	// TODO Discuss the use of multiple optional parameters
+	public async getTransactionCount<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		address: Address,
+		blockNumber: BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.getTransactionCount(
+			this._requestManager,
+			address,
+			blockNumber,
+		);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async sendTransaction(transaction: TransactionWithSender) {
+		return RpcMethods.sendTransaction(this._requestManager, transaction);
+	}
+
+	public async sendSignedTransaction(transaction: HexStringBytes) {
+		return RpcMethods.sendRawTransaction(this._requestManager, transaction);
+	}
+
+	public async sign(address: Address, message: HexStringBytes) {
+		return RpcMethods.sign(this._requestManager, address, message);
+	}
+
+	public async signTransaction(transaction: TransactionWithSender) {
+		return RpcMethods.signTransaction(this._requestManager, transaction);
+	}
+
+	public async call(
+		transaction: TransactionCall,
+		blockNumber: BlockNumberOrTag = this._options.defaultBlock,
+	) {
+		return RpcMethods.call(this._requestManager, transaction, blockNumber);
+	}
+
+	public async estimateGas<ReturnType extends ValidTypes = ValidTypes.HexString>(
+		transaction: Partial<TransactionWithSender>,
+		blockNumber: BlockNumberOrTag = this._options.defaultBlock,
+		returnType?: ReturnType,
+	) {
+		const response = await RpcMethods.estimateGas(
+			this._requestManager,
+			transaction,
+			blockNumber,
+		);
+
+		return convertToValidType(
+			response,
+			returnType ?? this._options.defaultReturnType,
+		) as ValidReturnTypes[ReturnType];
+	}
+
+	public async getPastLogs(filter: Filter) {
+		return RpcMethods.getLogs(this._requestManager, filter);
+	}
+
+	public async getWork() {
+		return RpcMethods.getWork(this._requestManager);
+	}
+
+	public async submitWork(
+		nonce: HexString8Bytes,
+		seedHash: HexString32Bytes,
+		difficulty: HexString32Bytes,
+	) {
+		return RpcMethods.submitWork(this._requestManager, nonce, seedHash, difficulty);
+	}
+
+	// TODO
+	// public async requestAccounts() {
+
+	// }
```

Comment:
- [1.x implementation](https://github.com/ChainSafe/web3.js/blob/3a3cb32b6434656131f0bfc4a8622130d19d2338/packages/web3-eth/src/index.js#L569)
- [1.x docs](https://web3js.readthedocs.io/en/v1.2.11/web3-eth.html#requestaccounts)
- Apart of [EIP-1102 ](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-1102.md)